src/main/java/com/faforever/client/filter/FilterMultiCheckboxController.java

```diff
@@ -0,0 +1,177 @@
+package com.faforever.client.filter;
+
+import com.faforever.client.fx.JavaFxUtil;
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.ui.list.NoFocusModelListView;
+import com.faforever.client.ui.list.NoSelectionModelListView;
+import com.google.common.annotations.VisibleForTesting;
+import javafx.beans.InvalidationListener;
+import javafx.beans.binding.Bindings;
+import javafx.beans.property.ListProperty;
+import javafx.beans.property.SimpleListProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ListChangeListener.Change;
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.FilteredList;
+import javafx.scene.control.MenuButton;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.VBox;
+import javafx.util.StringConverter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.controlsfx.control.CheckListView;
+import org.controlsfx.control.textfield.TextFields;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@RequiredArgsConstructor
+@Slf4j
+public class FilterMultiCheckboxController<U, T> extends AbstractFilterNodeController<List<U>, ListProperty<U>, T> {
+
+  public static int ITEM_AMOUNT_TO_ENABLE_SEARCH_BAR = 10;
+
+  private final I18n i18n;
+
+  public MenuButton root;
+  public VBox contentVBox;
+  public TextField searchTextField;
+  public CheckListView<String> listView;
+
+  private final ObservableList<U> selectedItems = FXCollections.observableArrayList();
+  private final ListProperty<U> property = new SimpleListProperty<>(selectedItems);
```

Comment:
```suggestion
  private final ObservableList<U> selectedItems = FXCollections.observableArrayList();
  private final ListProperty<U> selectedItemList = new SimpleListProperty<>(selectedItems);
```

Maybe a better name?